pkg/process/checks/container.go

```diff
@@ -119,6 +120,17 @@ func (c *ContainerCheck) Run(nextGroupID func() int32, options *RunOptions) (Run
 	if len(containers)%c.maxBatchSize != 0 {
 		groupSize++
 	}
+
+	// For no chunking, set groupsize as 1 to ensure one chunk
+	if options != nil && options.NoChunking {
+		oldGroupSize := groupSize
+		groupSize = 1
+
+		defer func() {
+			groupSize = oldGroupSize
+		}()
+	}
```

Comment:
Why can't the `groupSize` just be overwritten?

Reply:
True - groupSize is only used for the scope of the function, so defer is unnecessary. Will remove in next commits.